File: 01_Language/01_Functions/python/array_count_values.py
# coding: utf-8

def array_count_values(l):
    m = dict()
    for i in l:
        if m.get(i) is None:
            m[i] = 0
        m[i] += 1
    return m


# https://www.php2python.com/wiki/function.array-count-values/
def array_count_values2(l):
    from collections import defaultdict
    m = defaultdict(int)
    for i in l:
        m[i] += 1
    return m


# https://www.php2python.com/wiki/function.array-count-values/
def array_count_values3(l):
    from collections import Counter
    return Counter(l)


if __name__ == '__main__':
    array = [1, "hello", 1, "world", "hello"]
    print(array_count_values(array))
    print(array_count_values2(array))
    print(array_count_values3(array))

File: 01_Language/01_Functions/js/array_count_values.js
function array_count_values(arr) {
    let res = {}
    for (let i in arr) {
        let k = arr[i]
        if (res[k] === undefined) {
            res[k] = 0
        }
        res[k]++
    }
    return res
}

let array = [1, "hello", 1, "world", "hello"]
console.log(array_count_values(array));
